### ISSUE TITLE
Addon Manager: wrong processing of KUBECTL_EXTRA_PRUNE_WHITELIST

### ISSUE THREAD

**Issue by jkaniuk** [kubernetes/kubernetes#85343]
**What happened**:
Regression in 1.17 introduced by #82237
When there are several items passed using KUBECTL_EXTRA_PRUNE_WHITELIST flag,
cluster/addons/addon-manager/kube-addons.sh generates wrong flags for kubectl:

```
kubectl ... --prune-whitelist parameter1 parameter2 ...
```

Which causes:

```
error: Unexpected args: [parameter2]
See 'kubectl apply -h' for help and examples
```

**What you expected to happen**:
It should generate:

```
kubectl ... --prune-whitelist parameter1 --prune-whitelist parameter2 ...
```

**How to reproduce it (as minimally and precisely as possible)**:
```
export KUBECTL_EXTRA_PRUNE_WHITELIST=parameter1 parameter2
cluster/addons/addon-manager/kube-addons.sh
```

**Anything else we need to know?**:
/sig release

**Environment**:
- Kubernetes version (use `kubectl version`): 1.17.*

**Comment by wojtek-t** [comment]
@kubernetes/sig-release @guineveresaenger @alenkacz - FYI

This seems like a regression in 1.17, so we should merge fix in 1.17.

**Comment by liggitt** [comment]
cc @BenTheElder for shellcheck-induced regressions

**Comment by BenTheElder** [comment]
This happens because we don't have an "API" documented for these environment variables. Ideally we should not be depending on space splitting strings.

**Comment by jkaniuk** [comment]
Not yet fixed for 1.17

/reopen
